### PR TITLE
[CALCITE] Added support for HELP PROCEDURE statement

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlHelpProcedure.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+public class SqlHelpProcedure extends SqlHelp {
+  public static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("HELP PROCEDURE", SqlKind.HELP_PROCEDURE);
+
+  public final boolean attributes;
+
+  /**
+   * Creates a {@code SqlHelp}.
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlHelpProcedure.java:29]: nit: /SqlHelp/SqlHelpProcedure/

File: core/src/main/java/org/apache/calcite/sql/SqlHelpProcedure.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   * @param pos  Parser position, must not be null
+   * @param name  Name of the procedure
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlHelpProcedure.java:31]: nit: Missing "attributes" param doc.

File: parsing/intermediate/dialects/dialect1/parserImpls.ftl [hunk 3]
```diff
@@ -2653,6 +2653,8 @@ SqlNode SqlStmt() :
         stmt = SqlMerge()
     |
         stmt = SqlProcedureCall()
+    |
+        stmt = SqlHelp()
```
Comment (dasch-1) [inline parsing/intermediate/dialects/dialect1/parserImpls.ftl:2657]: nit: Order alphabetically.

File: parsing/intermediate/dialects/dialect1/parserImpls.ftl [hunk 4]
```diff
@@ -4517,3 +4519,34 @@ SqlCreateProcedureParameter SqlCreateProcedureParameter() :
...
+
+SqlHelpProcedure SqlHelpProcedure(Span s) :
+{
+    final SqlIdentifier procedureName;
+    boolean attributes = false;
+}
+{
+    <PROCEDURE> procedureName = CompoundIdentifier()
+    [
+        (<ATTRIBUTES> | <ATTR> | <ATTRS>) {
```
Comment (dasch-1) [inline parsing/intermediate/dialects/dialect1/parserImpls.ftl:4547]: nit: Missing spacing before "<ATTRIBUTES>" and after "<ATTRS>"

File: parsing/intermediate/dialects/dialect1/parserImpls.ftl (hunk 3 above)
Comment (michaelvdow) [reply parsing/intermediate/dialects/dialect1/parserImpls.ftl:2657]: Should I reorder the rest of the calls here since they are all out of order?

**Pull request merged**